src/passes/OptimizeInstructions.cpp

```diff
@@ -575,10 +575,17 @@ struct OptimizeInstructions
             }
           }
         }
-        // math operations on a constant power of 2 right side can be optimized
         if (right->type == Type::i32) {
           uint32_t c = right->value.geti32();
+          if (binary->op == DivUInt32 &&
+              c > uint32_t(std::numeric_limits<int32_t>::min()) &&
+              c < uint32_t(-1)) {
```

Comment:
Basically it could be simple `(int32_t)x < 0` but I want exclude `(unsigned)x / 0x80000000` and `(unsigned)x / -1` due to this special cases which handle differently by other rules and transform to `(unsigned)x >> 31` and `x != -1` accordantly.